spec/inputs/http_poller_spec.rb

```diff
@@ -2,87 +2,151 @@
+    describe "due to a non-existant host" do
+      let(:name) { default_name }
+      let(:url) { "thouetnhoeu89ueoueohtueohtneuohn" }
+      let(:code) { nil } # no response expected
 
-    it "should have a parse failut with not json" do
-      subject = LogStash::Inputs::HTTP_Poller.new(settings.merge({"urls" => {"url1" => "http://mytesturl.mydomain"}}))
-      subject.client.stub("http://mytesturl.mydomain", body: "my message is not valid json", code: 200)
+      let(:settings) { default_opts.merge("urls" => { name => url}) }
 
-      logstash_queue = Queue.new
-      subject.run_once logstash_queue
-      event = logstash_queue.pop
+      include_examples("unprocessable_requests")
+    end
 
-      expect(event["message"]["tags"]).to include("_jsonparsefailure")
-      expect(event["responseCode"]).to eq(200)
-      expect(event["success"]).to eq(true)
+    describe "due to a bogus port number" do
+      let(:name) { default_name }
+      let(:url) { "http://127.0.0.1:9999999" }
+      let(:settings) { default_opts.merge("urls" => {name => url}) }
+      let(:code) { nil } # No response expected
 
-      subject.teardown
+      include_examples("unprocessable_requests")
     end
+  end
+
+  describe "a codec mismatch" do
+    let(:inst) {
+      mod.new(default_opts)
+    }
+
+    let(:qmsg) {
+      inst.client.stub(default_url, body: "Definitely not JSON!", code: 200)
+      inst.send(:run_once, queue)
+      queue.pop(true)
+    }
 
-    it "should put the response in a hash" do
-      subject = LogStash::Inputs::HTTP_Poller.new(settings.merge({"urls" => {"url1" => "http://mytesturl.mydomain"}}))
-      subject.client.stub("http://mytesturl.mydomain", body: "{\"var1\":\"var1-value\"}", code: 200)
-
-      logstash_queue = Queue.new
-      subject.run_once logstash_queue
-      event = logstash_queue.pop
-      expect(event["responseCode"]).to eq(200)
-      expect(event["success"]).to eq(true)
-      expect(event["message"]["var1"]).to eq("var1-value")
-      subject.teardown
+    it "should send a _jsonparsefailure" do
+      expect(qmsg["tags"]).to include("_jsonparsefailure")
     end
+  end
 
-    it "should create 2 events" do
-      subject = LogStash::Inputs::HTTP_Poller.new(settings.merge({"urls" => {"url1" => "http://mytesturl.mydomain" ,"url2" => "http://mytesturl.mydomain2"}}))
-      subject.client.stub("http://mytesturl.mydomain", body: "{\"var1\":\"var1-value\"}", code: 200)
-      subject.client.stub("http://mytesturl.mydomain2", body: "{\"var2\":\"var2-value\"}", code: 200)
+  describe "a valid decoded response" do
+    let(:payload) {
+      {"a" => 2, "hello" => ["a", "b", "c"]}
+    }
+    let(:inst) {
+      mod.new(default_opts)
+    }
 
-      logstash_queue = Queue.new
-      subject.run_once logstash_queue
-      
-      expect{ logstash_queue.pop}.not_to raise_error
-      event = logstash_queue.pop
+    let(:qmsg) {
+      inst.client.stub(default_url, body: LogStash::Json.dump(payload), code: 202)
```

Comment:
hashrocket syntax here until https://github.com/elastic/logstash/pull/3485 is resolved